examples/medplum-demo-bots/src/dft-transfer-listener/dft-transfer-bot.ts

```diff
@@ -0,0 +1,141 @@
+/**
+ * This bot listens for DFT messages and finds/creates a Medplum Patient from PID,
+ * Coverage from IN1, and Claim from PR1.
+ *
+ * The bot will return an ACK message to the sender of the DFT message.
+ */
+import { BotEvent, createReference, Hl7Message, MedplumClient } from '@medplum/core';
+import { Claim, Coverage, Patient } from '@medplum/fhirtypes';
+
+export async function handler(medplum: MedplumClient, event: BotEvent<Hl7Message>): Promise<Hl7Message> {
+  const input = event.input;
+  const systemString = 'MRN';
+
+  // Verify message type is DFT
+  const messageType = input.getSegment('MSH')?.getField(9)?.getComponent(1) as string;
+  if (messageType !== 'DFT') {
+    throw new Error('Not a DFT message');
+  }
+
+  // Get patient information
+  const mrnNumber = input.getSegment('PID')?.getField(3)?.getComponent(1) as string;
+  const givenName = input.getSegment('PID')?.getField(5)?.getComponent(2) as string;
+  const familyName = input.getSegment('PID')?.getField(5)?.getComponent(1) as string;
+  const addressLine = input.getSegment('PID')?.getField(11)?.getComponent(1) as string;
+  const city = input.getSegment('PID')?.getField(11)?.getComponent(3) as string;
+  const state = input.getSegment('PID')?.getField(11)?.getComponent(4) as string;
+  const postalCode = input.getSegment('PID')?.getField(11)?.getComponent(5) as string;
+  const country = input.getSegment('PID')?.getField(11)?.getComponent(6) as string;
+
+  // Find or create patient
+  let patient = await medplum.searchOne('Patient', 'identifier=' + mrnNumber);
+  if (!patient) {
+    patient = await medplum.createResource<Patient>({
+      resourceType: 'Patient',
+      identifier: [
+        {
+          system: systemString,
+          value: mrnNumber,
+        },
+      ],
+      name: [
+        {
+          given: [givenName],
+          family: familyName,
+        },
+      ],
+      address: [
+        {
+          line: [addressLine],
+          city: city,
+          state: state,
+          postalCode: postalCode,
+          country: country,
+        },
+      ],
+    });
+  }
+
+  // Process insurance information if present
+  const in1Segment = input.getSegment('IN1');
+  let coverage: Coverage | undefined;
+
+  if (in1Segment) {
+    const insurerId = in1Segment.getField(3)?.getComponent(1) as string;
```

Comment:
1. With these, I prefer to add an inline comments, since lots of readers don't know the field/components off the top of your head

```suggestion
    const insurerId = in1Segment.getField(3)?.getComponent(1) as string;   // IN1.3 - Insurance Company ID
```

Reply:
Also why cast `as string`? Please handle `undefined` values